mobile-app/app/screens/AppNavigator/screens/Balances/components/BottomSheetAddressDetail.tsx

```diff
@@ -0,0 +1,313 @@
+import { RootState } from '@store'
+import { hasTxQueued } from '@store/transaction_queue'
+import { hasTxQueued as hasBroadcastQueued } from '@store/ocean'
+
+interface BottomSheetAddressDetailProps {
+  address: string
+  addressLabel: string
+  onReceiveButtonPress: () => void
+  onCloseButtonPress: () => void
+}
+
+export const BottomSheetAddressDetail = (props: BottomSheetAddressDetailProps): React.MemoExoticComponent<() => JSX.Element> => memo(() => {
+  const { isLight } = useThemeContext()
+  const flatListComponents = {
+    mobile: BottomSheetFlatList,
+    web: ThemedFlatList
+  }
+  const FlatList = Platform.OS === 'web' ? flatListComponents.web : flatListComponents.mobile
+  const { addressLength, setIndex, wallet, activeAddressIndex, discoverWalletAddresses } = useWalletContext()
+  const toast = useToast()
+  const [showToast, setShowToast] = useState(false)
+  const TOAST_DURATION = 2000
+  const [availableAddresses, setAvailableAddresses] = useState<string[]>([])
+  const [canCreateAddress, setCanCreateAddress] = useState<boolean>(false)
+  const logger = useLogger()
+  const dispatch = useDispatch()
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+
+  const onActiveAddressPress = useCallback(debounce(() => {
+    if (showToast) {
+      return
+    }
+    setShowToast(true)
+    setTimeout(() => setShowToast(false), TOAST_DURATION)
+  }, 500), [showToast])
+
+  useEffect(() => {
+    if (showToast) {
+      Clipboard.setString(props.address)
+      toast.show('Copied', {
+        type: 'wallet_toast',
+        placement: 'top',
+        duration: TOAST_DURATION
+      })
+    } else {
+      toast.hideAll()
+    }
+  }, [showToast, props.address])
+
+  // Getting addresses
+  const fetchAddresses = async (): Promise<void> => {
+    const addresses: string[] = []
+    for (let i = 0; i <= addressLength; i++) {
+      const account = wallet.get(i)
+      const address = await account.getAddress()
+      addresses.push(address)
+    }
+    setAvailableAddresses(addresses)
+    await isNextAddressUsable()
+  }
+
+  const isNextAddressUsable = async (): Promise<void> => {
+    // incremented 1 to check if next account in the wallet is usable.
+    const next = addressLength + 1
+    const isUsable = await wallet.isUsable(next)
+    setCanCreateAddress(isUsable && MAX_ALLOWED_ADDRESSES > next)
+  }
+
+  useEffect(() => {
+    fetchAddresses().catch(logger.error)
+  }, [wallet, addressLength])
+
+  useEffect(() => {
+    isNextAddressUsable().catch(logger.error)
+  }, [blockCount])
+
+  const CreateAddress = useCallback(() => {
+    if (!canCreateAddress) {
+      return <></>
+    }
+
+    return (
+      <ThemedTouchableOpacity
+        light={tailwind('bg-white border-gray-100')}
```

Comment:
This one is not that visible also on Dark mode. Should we consider changing? @jtlntno-des

Reply:
Will be tackled on a different issue